### ISSUE TITLE
Custom partitionioning "Decrypt" does not work

### ISSUE THREAD

**Issue by rasmus91** [elementary/installer#703]
### What Happened?

When installing with custom partitioning, upon selecting a LUKS partition, a prompt will pop up, asking to decrypt the partition, but when "decrypt" is pressed, nothing seems to happen.

### Steps to Reproduce

1. Go into demo mode
2. partition disk as: 
 - 500 MB fat32 for efi
 - XGB whatever
3. `cryptsetup luksFormat --type luks2 /dev/<partition>`
4. open the installer
5. When prompted for install type pick custom
6. click the luks partition to use it for something, enter password, and press decrypt

### Expected Behavior

I expected the drive to decrypt so that i could install root to it.

### OS Version

7.x (Horus)

### Software Version

Latest release (I have run all updates)

**Comment by neylsongularte** [comment]
In my case decrypt doesn't work either

![Captura de tela de 2024-03-24 11 53 11](https://github.com/elementary/installer/assets/4109823/4b9af53f-8715-4156-8dae-7a8c8c033b88)

**Comment by neylsongularte** [comment]
![image](https://github.com/elementary/installer/assets/4109823/c5eb5063-9f0c-468f-92e9-ac247f090c1e)

**Comment by rasmus91** [comment]
> In my case decrypt doesn't work either, there must be a problem when running on top of lvm

Well, I doubt that that's that, because I am not using LVM for my installs, and will not be using it.

**Comment by rasmus91** [comment]
So for elementary OS 8.0 it 'seems' to work a bit better.

I created a virtualbox VM and did the following:

1. boot elementary OS image
2. go into demo mode
3. create partitions in GParted: first create a new gpt partition table, then create the frist partition `/dev/sda1` which is just a small efi partition. (im not sure if a `/boot` is necessary) create a `/dev/sda2`
4. open a terminal and do the following (lets pretend your /root is `/dev/sda2`): 
```bash
sudo cryptsetup luksFormat --type luks2 /dev/sda2
# You will be prompted to make sure you mean it, type YES
sudo cryptsetup luksOpen /dev/sda2 crypt
# This creates /dev/mapper/crypt which is the raw luks2 partition, that now needs to be formatted
sudo mkfs.btrfs /dev/mapper/crypt
# Now theres a file system in it
```
5. Open the installer and proceed to disk layout, pick custom install. You will be able to click the `LUKS2` partition and the installer will prompt you for the name it should be mounted as, and the password, but you will not be able to select it as the install target

![image](https://github.com/user-attachments/assets/f2b1e4af-9b63-4ad0-ab8f-fbc57c1565f1)


**TL;DR**: The Installer will now decrypt and mount your luks2 partition, but it will not let you pick it as the `/root` or any other target.

**Comment by o-alquimista** [comment]
I cannot reproduce on elementary 8. When I type the password and press "Decrypt", a new disk appears below, containing the decrypted partition's LVM volumes; then I can click each LVM volume and use them for `/root`, `/home`, etc.

> I created a virtualbox VM and did the following...

Maybe it only fails inside a virtual machine? Have you tried it on the host system?

**Comment by jussihuotari** [comment]
I can reproduce on elementary 8.0.1 directly on a host system. After pressing "Decrypt", the installer decrypts and reports "LUKS volume was decrypted" but a new disk does not appear and the installation can't continue. FWIW this is on `ext4` logical volume (unlike OP's `btrfs`). If I go back in the installer and use the demo mode, I can see the logical volume in terminal as expected using `sudo lsblk -f`.
